cpp/non_iid_main.cpp

```diff
@@ -217,105 +217,101 @@ int main(int argc, char* argv[]) {
         H_bitstring = min(ret_min_entropy, H_bitstring);
+        tc631.ret_min_entropy = ret_min_entropy;
     }
 
     if (initial_entropy) {
         ret_min_entropy = most_common(data.symbols, data.len, data.alph_size, verbose, "Literal", tc631);
         if (verbose == 1) printf("\tMost Common Value Estimate = %f / %d bit(s)\n", ret_min_entropy, data.word_size);
         H_original = min(ret_min_entropy, H_original);
+        tc631.ret_min_entropy = ret_min_entropy;
     }
 
-    tc631.h_bitstring = H_bitstring;
-    tc631.h_original = H_original;
-    tc631.ret_min_entropy = ret_min_entropy;
-    tc631.data_word_size = data.word_size;
-    tc631.lrs_res = lrs_res;
     tc631.testCaseNumber = "Estimate entropy with Most Common Value";
     testRun.testCases.push_back(tc631);
 
     // Section 6.3.2 - Estimate entropy with Collision Test (for bit strings only)
+    NonIidTestCase tc632;
+
     if (verbose >= 1) printf("\nRunning Entropic Statistic Estimates (bit strings only)...\n");
 
     if (((data.alph_size > 2) || !initial_entropy)) {
         ret_min_entropy = collision_test(data.bsymbols, data.blen, verbose, "Bitstring");
         if (verbose == 1) printf("\tCollision Test Estimate (bit string) = %f / 1 bit(s)\n", ret_min_entropy);
         H_bitstring = min(ret_min_entropy, H_bitstring);
+        tc632.ret_min_entropy = ret_min_entropy;
     }
 
     if (initial_entropy && (data.alph_size == 2)) {
         ret_min_entropy = collision_test(data.symbols, data.len, verbose, "Literal");
         if (verbose == 1) printf("\tCollision Test Estimate = %f / 1 bit(s)\n", ret_min_entropy);
         H_original = min(ret_min_entropy, H_original);
+        tc632.ret_min_entropy = ret_min_entropy;
     }
 
-    NonIidTestCase tc632;
-    tc632.h_bitstring = H_bitstring;
-    tc632.h_original = H_original;
-    tc632.ret_min_entropy = ret_min_entropy;
-    tc632.data_word_size = data.word_size;
-    tc632.lrs_res = lrs_res;
     tc632.testCaseNumber = "Estimate entropy with Collision Test (for bit strings only)";
     testRun.testCases.push_back(tc632);
 
     // Section 6.3.3 - Estimate entropy with Markov Test (for bit strings only)
+    NonIidTestCase tc633;
+
     if (((data.alph_size > 2) || !initial_entropy)) {
         ret_min_entropy = markov_test(data.bsymbols, data.blen, verbose, "Bitstring");
         if (verbose == 1) printf("\tMarkov Test Estimate (bit string) = %f / 1 bit(s)\n", ret_min_entropy);
         H_bitstring = min(ret_min_entropy, H_bitstring);
+        tc633.ret_min_entropy = ret_min_entropy;
     }
 
     if (initial_entropy && (data.alph_size == 2)) {
         ret_min_entropy = markov_test(data.symbols, data.len, verbose, "Literal");
         if (verbose == 1) printf("\tMarkov Test Estimate = %f / 1 bit(s)\n", ret_min_entropy);
         H_original = min(ret_min_entropy, H_original);
+        tc633.ret_min_entropy = ret_min_entropy;
```

Comment:
Similarly, if both literal and bitstring assessments are conducted, only the literal assessment is saved.